### ISSUE TITLE
New folder doesn't show up in Nostalgy until Thunderbird restart

### ISSUE THREAD

**Issue by SKelem** [opto/nostalgy-xpi#200]
After I create a new folder in Thunderbird, that folder doesn't show up in a Nostalgy search until I restart Thunderbird.

I'm running Nostalgy 3.61 on Thunderbird 102.6.0.

**Comment by mdonz** [comment]
I experience similarly but it seems intermittent (TB 102.7.1, Nostalgy 3.6.13, Ubuntu 2.04) or dependent on order of activities?

I just restarted TB, then created a folder (under the root of the account) "test2" and then Nostalgy was able to show and use that folder. I then created folder "test3" as subfolder of an already existing folder amd Nostalgy does not see/offer this when trying to file an email and search for "test" (or use the "Go" command). I then created "test4" under the account root (where "test2" still is) and that is not visible to Nostalgy as well (until I restart TB and all three of these test folder are then usable.

**Comment by adamkerz** [comment]
I found that the second to last folder is known about, so if I create, in order:
 - folder1
 - folder2
 - folder3

Then Nostalgy++ knows about `folder1` and `folder2` but not `folder3` until another folder is created.

**Comment by adamkerz** [comment]
Update: looks like `folder3` appears after using nostalgy to make a move (eg. moving more stuff to `folder2` after the create `folder3` instruction was made).

**Comment by opto** [comment]
each time you move or copy or ..., we re-iterate the folders, starting from the server/account. I am not sure what happens if you abort/esc an action.

There could be a relation to folders being subscribed if in imap, I am not sure whether we see folders that are not subscribed, but only on the imap server.

So maybe it needs some time to propagate back from the imap server into the list?

**Comment by adamkerz** [comment]
Right, maybe that's it... The move to a new folder does seem to fail, since it doesn't exist yet.

Related: I put in a PR (I think for the original Nostalgy) where you could configure a delay between creating a new folder and the move being initiated, but my memory of the details are fuzzy now. Update, I found it: https://github.com/nostalgy/nostalgy/pull/24

Can @SKelem or @mdonz replicate this behaviour?

1. create new folder
1. attempt to select the new folder as a target of a move - expect that this fails, the folder is not listed
1. move an email to a different folder that nostalgy knows about
1. attempt the same move in 2. - expect that the folder is now listed

If my expectations are true, then perhaps @opto's hypothesis about imap folder subscriptions is the only thing going on here. 

Maybe a delay after creation but before moving AND before refreshing the folders is a solution?

**Comment by mdonz** [comment]
Earlier in this thread (February) I seemed to experience different behaviours. At the moment  (TB102.13.0, Nostalgy 3.6.18) I cannot replicate that issue. 
Creating folders makes them visible to Nostalgy immediately and I can use them i.e. as target for an email move.

**Comment by adamkerz** [comment]
😮 I just tried again and it's almost excellent! When I move to a non-existent folder, I get an error message that the folder doesn't exist, but then it's created and two seconds later, I can move to it again and it shows up. Yeah, this has changed at some point recently, but I don't know when. I'm pretty sure I was somewhat thorough two weeks ago when I tested, so has it changed since then? Something in Thunderbird 102.13?

**Comment by mdonz** [comment]
I was not even aware of the functionality that Nostalgy can create that folder at the time if needed. Interesting, I always manually create folders before ever using them.
I tested this here and the folder creation also works. I however then cannot use that folder soon after that creation with other Nostalgy commands like "g' to go that folder (although it is visible in my folder list) so my functionality seems slightly different to Adam's. At some stage later this folder is usable for Nostalgy commands but I have not spent time on figuring out when exactly.
All good for me.
